docs/developers/local-rules.md

```diff
@@ -0,0 +1,212 @@
+defined under the Scalafix Ivy configuration sources directory, which by
+default is `src/scalafix/scala`.
+
+```diff
+ repository
+ ├── build.sbt
+ ├── project
+ │   └── plugins.sbt
+ └── src
+     ├── main
+     │   └── ...
+     ├── test
+     │   └── ...
++    └── scalafix
++       ├── resources/META-INF/services
++       │   └── scalafix.v1.Rule // not mandatory, but allows syntax
++       |                        // `MyRule1` over `class:fix.MyRule1`
++       └── scala/fix
++           ├── MyRule1.scala
++           ├── MyRule2.scala
++           └── ...
+```
+
+```diff
+ // build.sbt
++libraryDependencies +=
++  "ch.epfl.scala" %%
++    "scalafix-core" %
++    _root_.scalafix.sbt.BuildInfo.scalafixVersion %
++    ScalafixConfig
+```
+
+```bash
+$ sbt
+> scalafix MyRule1 MyRule2
+```
+
+## Multi-project sbt build
+
+If your build contains several sub-projects, rules can be defined either:
+
+1. within [one of them](#within-a-given-sub-project), which is simpler but
+   make them impossible to test and complicate sharing rules across
+   sub-projects;
+1. under [their own sub-project(s)](#as-a-separate-sub-project), allowing
+   usage of [scalafix-testkit](setup.md).
+
+### Within a given sub-project
+
+In a large build with different code owners across sub-projects, it may
+be useful to define rules within a given sub-project, without impacting the
+global build definition. This helps to iterate on them before potentially
+promoting them to the [build-level](#as-a-separate-sub-project).
+
+```diff
+ repository
+ ├── build.sbt
+ ├── project
+ │   └── plugins.sbt
+ └── service1
+ │   └── src
+ │       ├── main
+ │       │   └── ...
+ │       ├── test
+ │       │   └── ...
++│       └── scalafix
++│           ├── resources/META-INF/services
++│           │   └── scalafix.v1.Rule // not mandatory, but allows syntax
++|           |                        // `MyRule1` over `class:fix.MyRule1`
++│           └── scala/fix
++│               ├── MyRule1.scala
++│               ├── MyRule2.scala
++│               └── ...
+ ├── service2
+ │   └── ...
+ └── service3
+     └── ...
+```
+
+```diff
+ // build.sbt
+ lazy val service1 = project
++  .settings(
++    libraryDependencies +=
++      "ch.epfl.scala" %%
++        "scalafix-core" %
++        _root_.scalafix.sbt.BuildInfo.scalafixVersion %
++        ScalafixConfig
++  )
+```
+
+```bash
+$ sbt
+> service1/scalafix MyRule1 MyRule2
+```
+
+### As a separate sub-project
+
+Declaring your rules in a separate sub-project allows you to use
+[scalafix-testkit](tutorial.md#write-unit-tests) to easily unit-test your
+semantic rules.
+
+```diff
+ repository
+ ├── build.sbt
+ ├── project
+ │   └── plugins.sbt
++├── scalafix // same source structure as the scalafix.g8 project template
++│   ├── input/src/main/scala/fix
++│   │   ├── Test1.scala
++│   │   ├── Test2.scala
++│   │   └── ...
++│   ├── output/src/main/scala/fix
++│   │   ├── Test1.scala
++│   │   ├── Test2.scala
++│   │   └── ...
++│   ├── rules/src
```

Comment:
this could be 
```suggestion
+│   ├── rules/scalafix
```
to potentially benefit from automatic addition of `scalafix-core`, but it would drift from the current template and would make the sbt `dependsOn` more awkward (`scalafix->scalafix`), so I decided to stick to it.